plugins/source_testing.go

```diff
@@ -21,14 +20,17 @@ type ResourceTestCase struct {
 	SkipIgnoreInTest bool
 }
 
-func init() {
-	_ = faker.SetRandomMapAndSliceMinSize(1)
-	_ = faker.SetRandomMapAndSliceMaxSize(1)
-}
-
-func TestSourcePluginSync(t *testing.T, plugin *SourcePlugin, logger zerolog.Logger, spec specs.Source) {
+func TestSourcePluginSync(t *testing.T, plugin *SourcePlugin, logger zerolog.Logger, spec specs.Source, opts ...TestSourcePluginOption) {
 	t.Helper()
 
+	o := &testSourcePluginOptions{}
+	for _, opt := range opts {
+		opt(o)
+	}
+	if !o.NoParallel {
```

Comment:
Last nit: Let's make this variable called `parallel` (also I think no need to export it so it can be lowercase).
default for `parallel: true` at line 26.

and let's do convention `WithTestSourcePluginNoParallel` - i.e `With` + `Name of the struct/option` + option